singer-io/tap-stripe
# TDL-19384 optimize logic for parent child relationship to be independent

# Description of change
- Optimized the parent-child relationship to fetch the parent only a single time in case when both are selected.
- Skip emitting the parent records when it is not selected
- Execute the sync function for the parent stream when the child stream is only selected
- Use the minimum bookmark across the selected streams and sub-streams

# Manual QA steps
 - Verify that the parent is fetched only once when both parent and child are selected.
- Verify that parent records are not emitted when it is not selected.
-  Verify that the minimum bookmark is used across the selected streams and sub-streams.

# Risks
 - 
 
# Rollback steps
 - revert this branch
